src/components/SignUpForm.tsx

```diff
@@ -42,54 +57,54 @@ const SignUpForm = () => {
 }
   return (
     <>
-    <form className='space-y-6 mt-20 p-3 flex flex-col justify-center relative'>
+    <form className='space-y-6 mt-20 p-3 flex flex-col justify-center relative' onSubmit={onSubmit}>
       {/* Input fields */}
     
         <input
+          {...register("name",{required:true})}
+          name='name'
           type='text'
-          value={name}
+          style={{borderColor: errors.name ? 'red' : ''}}
           placeholder='Full Name'
           className='border-b-2 outline-none'
-          onChange={(e) => setDetails({ name: e.target.value,email,password,confirmPassword })}
           />
-      
 
-  
         <input
+        {...register("email", {
+          required: true,
+          pattern: /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+        })}
+          name='email'
           type='text'
-          value={email}
+          style={{borderColor: errors.email ? 'red' : ''}}
           placeholder='Email'
           className='border-b-2 outline-none P-2'
-          onChange={(e) => setDetails({ email: e.target.value,name,password,confirmPassword })}
           />
-    
-          
+
     
         <input
+        {...register("password",{required:true, maxLength:18,minLength:6})}
+          name='password'
           type={showPassword ? 'text' : 'password'}
-          value={password}
           placeholder='Password'
           className='border-b-2 outline-none P-2'
-          onChange={(e) => setDetails({ password: e.target.value,name,email,confirmPassword })}
           />
       
-          
-    
+             
         <input
+        {...register("confirmPassword",{required:true, maxLength:18,minLength:6})}
+          name='confirmPassword'
           type={showConfirmPassword ? 'text' : 'password'}
-          value={confirmPassword}
-          placeholder='Confirm Password'
-          className='border-b-2 outline-none P-2'
-          onChange={(e) => setDetails({ password,name,email,confirmPassword: e.target.value })}
-          
+          placeholder='Confirm password'
+          className='border-b-2 outline-none P-2'   
           />
```

Comment:
it seems you are not showing validation messages.....one way of doing that..is simply conditionally rendering a message
if password && confirm password are not the same